src/cloudant/account.py

```diff
@@ -392,32 +392,34 @@ def _usage_endpoint(self, endpoint, year=None, month=None):
         optional year and month URL elements.
 
         :param str endpoint: Cloudant usage endpoint.
-        :param int year: Year to query against.  Defaults to None.
-        :param int month: Month to query against.  Defaults to None.
-        """
-        if year is not None:
-            endpoint = posixpath.join(endpoint, str(year))
-        if month is not None:
-            if year is None:
-                raise CloudantException(
-                    (
-                        "must supply both year and month "
-                        "to usage endpoint: {0}"
-                    ).format(endpoint)
-                )
-            endpoint = posixpath.join(endpoint, str(month))
+        :param int year: Year to query against.  Optional parameter.
+        Defaults to None.  If used, it must be accompanied by the ``month``.
+        :param int month: Month to query against. Optional parameter.
+        Defaults to None.  If used, it must be accompanied by the ``year``.
+        """
+        if (year is not None) and (month is not None):
+            endpoint = posixpath.join(endpoint, str(year), str(month))
```

Comment:
@mikerhodes: I think using `posixpath` is OK to use in [constructing URLs](http://effbot.org/librarybook/posixpath.htm).  I've also checked the posixpath.py docstring in both Python 2.7 and 3.5 and they both contain:

```
...
Some of this can actually be useful on non-Posix systems too, e.g.
for manipulation of the pathname component of URLs.
```